### PR TITLE
[lxd] Add try/catch around updating and pruning images

### LINKED ISSUE #1689: [lxd] daemon exits with "unhandled exception: LXD object not found" when starting with no multipass project defined
The daemon will bail on start up when there is no `multipass` project defined yet.

### PR DESCRIPTION
**Pull request by townsend2010**
Previously, when the "multipass" project didn't exist on daemon start, there would be an unhandled exception and the daemon would exit.

Fixes #1689

### CODE REVIEW

File: src/platform/backends/lxd/lxd_vm_image_vault.cpp [hunk 1]
```diff
@@ -195,63 +195,75 @@ bool mp::LXDVMImageVault::has_record_for(const std::string& name)
...
+                lxd_request(
+                    manager, "DELETE",
+                    QUrl(QString("%1/images/%2").arg(base_url.toString()).arg(image_info["fingerprint"].toString())));
+            }
         }
     }
+    catch (const LXDNotFoundException&)
+    {
+        return;
+    }
```
Comment (Saviq) [inline src/platform/backends/lxd/lxd_vm_image_vault.cpp:224]: This feels a bit broad… If the `DELETE` request returns 404, should we still just ignore it?

File: src/platform/backends/lxd/lxd_vm_image_vault.cpp [hunk 2]
```diff
@@ -195,63 +195,75 @@ bool mp::LXDVMImageVault::has_record_for(const std::string& name)
...
 
-            poll_download_operation(json_reply, monitor, task_complete);
+                poll_download_operation(json_reply, monitor, task_complete);
+            }
         }
     }
+    catch (const LXDNotFoundException&)
+    {
+        return;
+    }
```
Comment (Saviq) [inline src/platform/backends/lxd/lxd_vm_image_vault.cpp:266]: Ditto? There's a lot more requests than just the update one that you're catching here, and not all of them should be ignored, I'd say?

File: src/platform/backends/lxd/lxd_vm_image_vault.cpp (hunk 1 above)
Comment (townsend2010) [reply src/platform/backends/lxd/lxd_vm_image_vault.cpp:224]: Right, I was too focused on the first `lxd_request()`.  Will work on a better solution.

Comment (townsend2010) [reply src/platform/backends/lxd/lxd_vm_image_vault.cpp]: Ok, this should be better now.

File: src/platform/backends/lxd/lxd_vm_image_vault.cpp (hunk 2 above)
Comment (townsend2010) [reply src/platform/backends/lxd/lxd_vm_image_vault.cpp]: Should be better now.

**Pull request merged**